packages/flutter/lib/src/widgets/scroll_view.dart

```diff
@@ -1110,13 +1110,15 @@ class ListView extends BoxScrollView {
   /// The `itemBuilder` callback will be called only with indices greater than
   /// or equal to zero and less than `itemCount`.
   ///
-  /// The `itemBuilder` should always return a non-null widget, and actually
-  /// create the widget instances when called. Avoid using a builder that
-  /// returns a previously-constructed widget; if the list view's children are
-  /// created in advance, or all at once when the [ListView] itself is created,
-  /// it is more efficient to use the [ListView] constructor. Even more
-  /// efficient, however, is to create the instances on demand using this
-  /// constructor's `itemBuilder` callback.
+  /// It is legal for `itemBuilder` to return `null`. If it does, [ListView]
```

Comment:
I am still not certain this is unintentional, especially since it is documented.

It was possible to return null in the past before null safety because there was no null safety.
It looks like GridView.builder also does not allow null.

Reply:
I've updated GridView.builder too, which I think has the same issue

Git says you wrote the documentation about this, so you're a far better source of truth than me.

But supporting null items is too simple to implement for it not to be voluntarily supported IMO.
This PR only modifies types after all. It doesn't modify logic in any way

Reply:
But considering this behavior is 2 years old now, it probably doesn't matter if it was intentional or not. Even if it was unintentionally removed, people don't seem to mind.

So maybe an alternate way of moving this forward would be to consider that this was never a feature to begin with, and to consider this PR a feature request.

Reply:
Thanks! I am ok moving forward with this change based on our digging. It may be worth documenting the trade offs (or say, expected differences in behavior) of a `return null` strategy versus an item count.

Can you see if there are other cases of `IndexedItemBuilder` that need to be updated?